packages/engine/paima-sm/src/cde-evm-batcher-payment.ts

```diff
@@ -0,0 +1,35 @@
+import type { PoolClient } from 'pg';
+
+import { doLog } from '@paima/utils';
+import type { CdeBatcherPaymentDatum } from './types.js';
+import type { SQLUpdate } from '@paima/db';
+import { cdeBatcherPaymentUpdateBalance } from '@paima/db';
+
+export default async function processBatcherPayment(
+  readonlyDBConn: PoolClient,
+  cdeDatum: CdeBatcherPaymentDatum,
+  inPresync: boolean
+): Promise<SQLUpdate[]> {
+  const cde_name = cdeDatum.cdeName;
+  const { batcherAddress, userAddress, value } = cdeDatum.payload;
+
+  const updateList: SQLUpdate[] = [];
+  console.log('processBatcherPayment', cdeDatum, inPresync);
+  try {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    updateList.push([
+      cdeBatcherPaymentUpdateBalance,
+      {
+        cde_name: 'generic-batcher-contract', // TODO Using constant name. Can there be more than one?
```

Comment:
What to do with this.

Reply:
I don't understand why is this a concern. There is a cdeName in the datum that can be used here. Which I imagine comes from the config. Even if there can be only one entry of this extension type for some reason, we should probably validate that instead.